### ISSUE TITLE
Xenosaga Episode II [NTSC-U]: Minor icon/font rendering issues

### ISSUE THREAD

**Issue by trostboot** [PCSX2/pcsx2#1883]
***PCSX2 version:***
1.5.0-dev-1976

***PCSX2 options:***
defaults

***Plugins used:***
GSdx DX11 SW/HW, OGL SW/HW

***Description of the issue:***
Icons rendered in text boxes have some extra black lines and edges. Most noticeably with the square icon.
PCSX2:
![gsdx_20170403222945](https://cloud.githubusercontent.com/assets/17594477/24631647/68bfd256-18c1-11e7-98ac-042ba62b88bc.png)
Real PS2 (sorry for the ghetto CRT picture, can't capture PS2 on PC right now):
![xs2icons](https://cloud.githubusercontent.com/assets/17594477/24631705/97e2f7ca-18c1-11e7-8a0c-7a363a5634af.jpg)
Smudginess aside, I think it's still fairly obvious.
Filtering options have no effect on it. Forced everything just introduces various other issues (as might be expected).
GSdump: 
[gsdx_20170403222945.zip](https://github.com/PCSX2/pcsx2/files/891423/gsdx_20170403222945.zip)

More generally, and perhaps related, font rendering in XS2 seems ever so slightly off. It looks a lot better on a real PS2, but I don't know how much of that might be down to the CRT smoothing over some of the rough edges. I'd feel more confident about this if I could capture directly from the PS2. 

***Last known version to work:***
Present as far back as 1.0.0 and 0.9.6

***PC specifications:***
i5-4690k, R9 270X, 16GB RAM, Win10 Pro 14393 x64

**Comment by FlatOutPS2** [comment]
Judging by the GSdump, you can fix it with the "Wild Arms Offset" hack when using a hardware renderer.

**Comment by trostboot** [comment]
Indeed, that might be helpful to know. I wouldn't have looked at that, as it requires upscaling, which introduces so many issues in XS2 as to not be worth it.

**Comment by gregory38** [comment]
First how is it working with the SW renderer. Did you try to enable aa1 ?

**Comment by trostboot** [comment]
It's the same across all renderers, both HW and SW. Edge AA has no effect on it.

**Comment by FlatOutPS2** [comment]
> it's greyed out at native res, but will still be active when selected prior to switching to native

It seems that's been the case ever since it was greyed out for native. In v1.4.0 and earlier it was available for native resolution too.

@gregory38 I didn't find any setting that fixed the issue for SW and HW other than the one I mentioned.

**Comment by gregory38** [comment]
Hum it feels like an upscaling issue. Interesting. It could be a float accuracy issue. The real HW will trunc the mantissa of `S/T/Q` values somewhere in the rasterization process. However current implementation uses the full mantissa range. Which mean that texture coordinate is slightly bigger than real HW. And so we sample the next texel (aka the next texture) hence the strange border.

**Comment by MrCK1** [comment]
@trostboot Any status update on this? Would be good to check latest git.

**Comment by trostboot** [comment]
Same as before, as you'd expect.

**Comment by Mrlinkwii** [comment]
is this still an issue on latest dev , if it is  can you post an updated  GS dump ? :slightly_smiling_face:

**Comment by coornio** [comment]
Tested the dump in the original message. The issue is indeed still present to this day, occurs both on hardware AND software rendering.

For hardware rendering, the issue can be eliminated through the Wild Arms Offset hack (fixes the incorrect icon scaling in dialogs) and Round Sprite: Full (fixes the odd beige vertical lines that appear on the right screen edge). Understandably, these fixes can only take place in an upscaled scenario. From the gamedb I can see that there's other hacks involved as well so I'll try and talk to someone who owns the title about them.

**Comment by Jakey757** [comment]
anyone check this with https://github.com/PCSX2/pcsx2/pull/6553?
